doc/lsst.pipe.base/creating-a-command-line-task.rst

```diff
@@ -0,0 +1,253 @@
+         return parser
+
+  In this case the custom container class `~lsst.pipe.tasks.coaddBase.SelectDataIdContainer` adds additional information for the task, to save processing time.
+
+- A task's `run` method requires no data references at all.
+  An example is ``makeSkyMap.MakeSkyMapTask``, which makes a sky map for a set of co-adds.
+  ``makeSkyMap.MakeSkyMapTask._makeArgumentParser`` is trivial:
+
+  .. code-block:: python
+
+     @classmethod
+     def _makeArgumentParser(cls):
+         """Create an argument parser
+         No identifiers are added because none are used.
+         """
+         return pipeBase.ArgumentParser(name=cls._DefaultName)
+
+.. _creating-a-command-line-task-custom-task-runner:
+
+Custom Task Runner
+==================
+
+The standard task runner is `lsst.pipe.base.TaskRunner`.
+It assumes that your task's ``run`` method wants a single data reference and nothing else.
+If that is not the case then you will have to provide a custom task runner for your task.
+This involves writing a subclass of `lsst.pipe.base.TaskRunner` and specifying it in your task using the ``RunnerClass`` :ref:`class variable <creating-a-task-class-variables>`.
+
+Here are some situations where a custom task runner is required:
+
+- The task's ``run`` method requires extra arguments.
+  An example is co-addition, which optionally accepts a list of images to co-add.
+  The custom task runner is ``coaddBase.CoaddTaskRunner`` and is pleasantly simple:
+
+  .. code-block:: python
+
+     class CoaddTaskRunner(pipeBase.TaskRunner):
+         @staticmethod
+         def getTargetList(parsedCmd, **kwargs):
+             return pipeBase.TaskRunner.getTargetList(parsedCmd, selectDataList=parsedCmd.selectId.dataList,
+                                                      **kwargs)
+
+- The task requires no data references, just a butler.
+  An example is ``makeSkyMap.MakeSkyMapTask``, which makes a ``skymap.SkyMap`` for a set of co-adds.
+  It uses the custom task runner ``makeSkyMap.MakeSkyMapRunner``, which is more complicated than the previous example because the entire ``__call__`` method must be overridden:
+
+  .. code-block:: python
+
+     class MakeSkyMapRunner(pipeBase.TaskRunner):
+         """Only need a single butler instance to run on."""
+         @staticmethod
+         def getTargetList(parsedCmd):
+             return [parsedCmd.butler]
+         def __call__(self, butler):
+             task = self.TaskClass(config=self.config, log=self.log)
+             if self.doRaise:
+                 results = task.run(butler)
+             else:
+                 try:
+                     results = task.run(butler)
+                 except Exception as e:
+                     task.log.fatal("Failed: %s" % e)
+                     if not isinstance(e, pipeBase.TaskError):
+                         traceback.print_exc(file=sys.stderr)
+             task.writeMetadata(butler)
```

Comment:
Is this up-to-date with the boiler-plate changes needed to return correct exit codes to the shell?